src/serialization/mod.rs

```diff
@@ -80,23 +80,43 @@ impl<T: CipherSuite> From<&LoginThirdMessage<T>> for ProtocolMessageType {
     }
 }
 
-pub(crate) fn serialize(input: &[u8], max_bytes: usize) -> Vec<u8> {
-    let mut output: Vec<u8> = Vec::new();
-    output.extend_from_slice(&input.len().to_be_bytes()[8 - max_bytes..]);
-    output.extend_from_slice(&input[..]);
+// Corresponds to the I2OSP() function from RFC8017
+pub(crate) fn i2osp(input: usize, length: usize) -> Vec<u8> {
+    if length <= 8 {
+        return (&input.to_be_bytes()[8 - length..]).to_vec();
```

Comment:
If usize is 32 bytes on a particular platform, and I want to write it on len=4
bytes, it seems I'd be writing .. nothing? `std::mem::size_of::<usize>()` may help here.

Reply:
Changed "8" to std::mem::size_of::<usize>() everywhere